spec/oauth/oidc_spec.lua

```diff
@@ -227,6 +228,58 @@ describe('OIDC', function()
       local credentials, _, _, err = oidc:transform_credentials({ access_token = access_token })
       assert(credentials, err)
     end)
+
+    describe('with app_id modification', function()
+
+      it('validation of app_id template', function()
+        stub(ngx_variable, 'available_context', function(context) return context end)
+
+        local oidc = _M.new(oidc_config, {oidc={client_id="{{aud}}"}})
+        local access_token = jwt:sign(rsa.private, {
+          header = { typ = 'JWT', alg = 'RS256', kid = 'somekid' },
+          payload = {
+            iss = oidc_config.issuer,
+            aud = 'notused',
+            azp = 'ce3b2e5e',
+            sub = 'someone',
+            exp = ngx.now() + 10,
+          },
+        })
+
+        local credentials, ttl, _, err = oidc:transform_credentials({ access_token = access_token })
+
+        assert(credentials, err)
+
+        assert.same({ app_id  = "notused" }, credentials)
+        assert.equal(10, ttl)
+      end)
+
+      it('validation of app_id template with functions', function()
+        stub(ngx_variable, 'available_context', function(context) return context end)
```

Comment:
This appears in both tests. We could avoid the duplication by extracting this to a `before` function.